src/index.ts

```diff
@@ -147,9 +148,19 @@ function setModuleMeta(
     // a previously cached version the TypeScript may be different and therefore should be
     // treated as new
     loaderContext._module!.buildMeta.tsLoaderFileVersion = fileVersion;
+    const impliedNodeFormat = getImpliedNodeFormat(
+      loaderContext._module!.resource,
```

Comment:
Please use `loaderContext.resourcePath`, resource can contains `?foo=bar` and `#hash`